cdap-authorization-client/src/main/java/co/cask/cdap/authorization/ACLManagerClient.java

```diff
@@ -0,0 +1,198 @@
+    }
+
+    return ObjectResponse.fromJsonBody(response, new TypeToken<Set<ACLEntry>>() { }).getResponseObject();
+  }
+
+  public Set<ACLEntry> listACLs() throws IOException {
+    String path = "/v1/acls";
+    HttpRequest request = HttpRequest.get(resolveURL(path)).addHeaders(getHeaders()).build();
+    HttpResponse response = HttpRequests.execute(request);
+
+    if (response.getResponseCode() != HttpURLConnection.HTTP_OK) {
+      throw new IOException("Unexpected response: " + response.getResponseCode() +
+                              ": " + response.getResponseMessage());
+    }
+
+    return ObjectResponse.fromJsonBody(response, new TypeToken<Set<ACLEntry>>() { }).getResponseObject();
+  }
+
+  public void deleteGlobalACLs(ACLStore.Query query) throws IOException {
+    String path = appendQuery("/v1/acls/global", query);
+    HttpRequest request = HttpRequest.delete(resolveURL(path)).addHeaders(getHeaders()).build();
+    HttpResponse response = HttpRequests.execute(request);
+
+    if (response.getResponseCode() != HttpURLConnection.HTTP_OK) {
+      throw new IOException("Unexpected response: " + response.getResponseCode() +
+                              ": " + response.getResponseMessage());
+    }
+  }
+
+  public void deleteACLs(String namespaceId, ACLStore.Query query) throws IOException {
+    String path = appendQuery("/v1/acls/namespace/" + namespaceId, query);
+    HttpRequest request = HttpRequest.delete(resolveURL(path)).addHeaders(getHeaders()).build();
+    HttpResponse response = HttpRequests.execute(request);
+
+    if (response.getResponseCode() != HttpURLConnection.HTTP_OK) {
+      throw new IOException("Unexpected response: " + response.getResponseCode() +
+                              ": " + response.getResponseMessage());
+    }
+  }
+
+  /**
+   * Creates an {@link ACLEntry} in a namespace for an object, subject, and a permission.
+   * This allows the subject to access the object for the specified permission.
+   *
+   * <p>
+   * For example, if object is "secretFile", subject is "Bob", and permission is "WRITE", then "Bob"
+   * would be allowed to write to the "secretFile", assuming that what is doing the writing is protecting
+   * the "secretFile" via a call to one of the {@code verifyAuthorized()} or {@code isAuthorized()} calls.
+   * </p>
+   *
+   * @param entry the {@link ACLEntry} to create
+   * @throws java.io.IOException if an error occurred when contacting the authorization service
+   */
+  public void createACL(String namespaceId, ACLEntry entry) throws IOException {
+    HttpRequest request = HttpRequest.post(resolveURL("/v1/acls/namespace/" + namespaceId))
+      .withBody(GSON.toJson(entry)).addHeaders(getHeaders()).build();
+    HttpResponse response = HttpRequests.execute(request);
+
+    if (response.getResponseCode() != HttpURLConnection.HTTP_OK &&
+      response.getResponseCode() != HttpURLConnection.HTTP_NOT_MODIFIED) {
```

Comment:
When will a service return `HTTP_NOT_MODIFIED`? `HTTP_NOT_MODIFIED` is for conditional `GET`, right?